crates/consensus-logic/src/csm/worker.rs

```diff
@@ -36,140 +26,91 @@ pub struct WorkerState {
-
-        let (cur_state_idx, cur_state) = csman
-            .get_most_recent_state_blocking()
-            .ok_or(Error::MissingClientState(0))?;
-
-        // TODO make configurable
-        let config = CsmExecConfig {
-            retry_base_dur: time::Duration::from_millis(1000),
-            // These settings makes the last retry delay be 6 seconds.
-            retry_cnt_max: 20,
-            retry_backoff_mult: 1120,
-        };
+    pub fn open(params: Arc<Params>, storage: Arc<NodeStorage>) -> anyhow::Result<Self> {
+        let (cur_block, cur_state) = storage
+            .client_state()
+            .fetch_most_recent_state()?
+            .expect("missing initial client state?");
 
         Ok(Self {
             params,
-            config,
             storage,
-            cur_state_idx,
-            cur_state,
-            cupdate_tx,
-            checkpoint_manager,
+            cur_state: Arc::new(cur_state),
+            cur_block,
         })
     }
 
-    /// Gets the index of the current state.
-    pub fn cur_event_idx(&self) -> u64 {
-        self.cur_state_idx
-    }
-
     /// Gets a ref to the consensus state from the inner state tracker.
     pub fn cur_state(&self) -> &Arc<ClientState> {
         &self.cur_state
     }
 
-    /// Gets a reference to checkpoint manager
-    pub fn checkpoint_db(&self) -> &CheckpointDbManager {
-        self.checkpoint_manager.as_ref()
+    pub fn cur_block(&self) -> L1BlockCommitment {
+        self.cur_block
     }
 
-    fn get_sync_event(&self, idx: u64) -> anyhow::Result<Option<SyncEvent>> {
-        Ok(self.storage.sync_event().get_sync_event_blocking(idx)?)
-    }
-
-    /// Fetches a sync event from storage.
-    fn get_sync_event_ok(&self, idx: u64) -> anyhow::Result<SyncEvent> {
-        Ok(self
-            .get_sync_event(idx)?
-            .ok_or(Error::MissingSyncEvent(idx))?)
-    }
-
-    /// Given the next event index, computes the state application if the
-    /// requisite data is available.  Returns the output and the new state.
-    ///
-    /// This is copied from the old `StateTracker` type which we removed to
-    /// simplify things.
-    // TODO maybe remove output return value
-    pub fn advance_consensus_state(&mut self, ev_idx: u64) -> anyhow::Result<ClientUpdateOutput> {
-        let prev_ev_idx = ev_idx - 1;
-        if prev_ev_idx != self.cur_state_idx {
-            return Err(Error::SkippedEventIdx(prev_ev_idx, self.cur_state_idx).into());
-        }
-
-        // Load the event from the database.
-        let ev = self.get_sync_event_ok(ev_idx)?;
-
-        debug!(%ev_idx, ?ev, "processing sync event");
+    /// Given the next l1 block, does the state transition, returning next [`ClientState`].
+    pub fn advance_consensus_state(
+        &self,
+        next_block: &L1BlockManifest,
+    ) -> anyhow::Result<(ClientState, Vec<SyncAction>)> {
+        let id = next_block.blkid();
+        debug!(%id, "processing l1 block");
```

Comment:
This feels like it could benefit from having sanity checks about blocks following from their parents, but this might be refactored in a subsequent PR and seems to work now, so this can probably be left as-is for now.